packages/aws-cdk-lib/aws-cloudwatch/lib/variable.ts

```diff
@@ -0,0 +1,231 @@
+   */
+  public static fromSearchComponents(namespace: string, dimensions: string[], metricName: string, populateFrom: string): Values {
+    if (dimensions.length === 0) {
+      throw new Error('Empty dimensions provided. Please specify one dimension at least');
+    }
+    if (!dimensions.includes(populateFrom)) {
+      throw new Error(`populateFrom (${populateFrom}) is not present in dimensions`);
+    }
+    const components = [namespace, ...dimensions];
+    return new SearchValues(`{${components.join(',')}} MetricName=\"${metricName}\"`, populateFrom);
+  }
+
+  /**
+   * Create values from a search expression
+   *
+   * @param expression search expression that specifies a namespace, dimension name(s) and a metric name. For example `{AWS/EC2,InstanceId} MetricName=\"CPUUtilization\"`
+   * @param populateFrom dimension the dimension name, that the search expression retrieves, whose values will be used to populate the values to choose from. For example `InstanceId`
+   */
+  public static fromSearch(expression: string, populateFrom: string): Values {
+    return new SearchValues(expression, populateFrom);
+  }
+
+  /**
+   * Create values from an array of possible variable values
+   */
+  public static fromValues(...values: VariableValue[]): Values {
+    if (values.length == 0) {
+      throw new Error('Empty values is not allowed');
+    }
+    return new StaticValues(values);
+  }
+
+  public abstract toJson(): any;
+}
+
+class StaticValues extends Values {
+  constructor(private readonly values: VariableValue[]) {
+    super();
+  }
+
+  toJson(): any {
+    return {
+      values: this.values.map(value => ({ label: value.label, value: value.value })),
+    };
+  }
+}
+
+class SearchValues extends Values {
+  /**
+   * Create a search expression for use
+   *
+   * @param expression search expression that specifies a namespace, dimension name(s) and a metric name. For example `{AWS/EC2,InstanceId} MetricName=\"CPUUtilization\"`
+   * @param populateFrom dimension the dimension name, that the search expression retrieves, whose values will be used to populate the values to choose from. For example `InstanceId`
+   */
+  public constructor(public readonly expression: string, public readonly populateFrom: string) {
+    super();
+  }
+
+  toJson(): any {
+    return {
+      search: this.expression,
+      populateFrom: this.populateFrom,
+    };
+  }
+}
+
+/**
+ * Default value for use in {@link DashboardVariableOptions}
+ */
+export class DefaultValue {
+
+  /**
+   * A special value for use with search expressions to have the default value be the first value returned from search
+   */
+  public static readonly FIRST = new DefaultValue('__FIRST');
+
+  /**
+   * Create a default value
+   * @param value the value to be used as default
+   */
+  public static of(value: any) {
+    return new DefaultValue(value);
+  }
```

Comment:
I don't love `of`. Reason being that we're not creating a default value 'of' something. We're just referencing a value. Some options:

```ts
DefaultValue.string('us-east-1')
DefaultValue.value('us-east-1') // contributes to overkill of the word value though
```

I don't really care. I just dont think `of` is right.

Reply:
I honestly can't think of a better option than `of` - except maybe the overkill `value` you suggested :)
`string` doesn't align well with the fact that it's actually `any`, with the documentation saying `Type: String, Number, or Boolean, depending on the type value for this variable`

I think I'll go with `value`